<launch>
  <!--include file="$(find exploration)/launch/joystick_control.launch" /-->
  <include file="$(find bml_vrep)/launch/mapping.launch" />
  
  <node name='active_pub' pkg='rostopic' type='rostopic' args='pub -r 10 /active std_msgs/Bool "data: true"' />
  <node name='marker_aggregator' pkg='exploration' type='marker_aggregator.py' args="0.0 1.0"/>

  <group ns="robot_00">
    <node name="add_moving_imu" pkg="coop_loc" type="add_moving_imu.py" />
    <node name="pose_controller" pkg="exploration" type="pose_controller.py" />
    <node name='mocap_pose_estimator' pkg='exploration' type='mocap_pose_estimator.py' />
    <!--node name="pose_planner" pkg="exploration" type="pose_planner.py" /-->
    <group ns="camera_0">
      <include file="$(find monocular_pose_estimator)/launch/factored.launch" />
    </group>
    <group ns="markers_0">
      <node name="marker_controller" pkg="exploration" type="marker_controller.py" />
      <!--node name="marker_planner" pkg="exploration" type="marker_planner.py" /-->
    </group>
    <group ns="scanner_0">
      <node name="scan_controller" pkg="exploration" type="scan_controller.py" />
      <!--node name="scan_planner" pkg="exploration" type="scan_planner.py" /-->
    </group>
  </group>

  <group ns="robot_01">
    <node name="add_moving_imu" pkg="coop_loc" type="add_moving_imu.py" />
    <node name="pose_controller" pkg="exploration" type="pose_controller.py" />
    <node name='mocap_pose_estimator' pkg='exploration' type='mocap_pose_estimator.py' />
    <!--node name="pose_planner" pkg="exploration" type="pose_planner.py" /-->
    <group ns="camera_0">
      <include file="$(find monocular_pose_estimator)/launch/factored.launch" />
    </group>
    <group ns="markers_0">
      <node name="marker_controller" pkg="exploration" type="marker_controller.py" />
      <!--node name="marker_planner" pkg="exploration" type="marker_planner.py" /-->
    </group>
    <group ns="scanner_0">
      <node name="scan_controller" pkg="exploration" type="scan_controller.py" />
      <!--node name="scan_planner" pkg="exploration" type="scan_planner.py" /-->
    </group>
  </group>
</launch>
